### PR TITLE
add a contributing doc

### PR DESCRIPTION
**Pull request by goern**
basically ported it from the template project

Signed-off-by: Christoph Görn <goern@redhat.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+3. Once you have completed your work, create a Pull Request, ensuring that it meets the requirements listed below.
+
+### Requirements for Pull Requests (PR)
+
+- Use `pre-commit` (see [below](#checkerslintersformatters--pre-commit)).
+- Use common sense when creating commits, not too big, not too small. You can also squash them at the end of review. See [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/).
+- Cover new code with a test case (new or existing one).
+- All tests have to pass.
+- Rebase against updated `master` branch before creating a PR to have linear git history.
+- Create a PR against the `master` branch.
```
Comment (KPostOffice) [inline CONTRIBUTING.md:51]: Can we add a line here about running mypy in an appropriate venv (if applicable)?

Comment (goern) [reply CONTRIBUTING.md:51]: isnt that mentioned a few lines below?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+Please take a few minutes to read GitHub's guide on [How to Contribute to Open Source](https://opensource.guide/how-to-contribute/).
+It's a quick read, and it's a great way to introduce yourself to how things work behind the scenes in open-source projects.
+
+### Dependencies
+
+If you are introducing a new dependency, please make sure it's added to:
+
+- `Pipfile`
+- `setup.py` if its a package
```
Comment (pacospace) [inline CONTRIBUTING.md:32]: don't we place it in`requirements.txt` and read it from setup.py?

Comment (goern) [reply CONTRIBUTING.md:32]: @fridex what is the good practice that we want here?

Comment (fridex) [reply CONTRIBUTING.md:32]: We use `requirements.txt` files for library releases (requirements on direct dependencies) and Pipfile/Pipfile.lock for components (fully pinned down stack):

- Pipfile/Pipfile.lock if the repo is a component built by Python s2i
- `requirements.txt` (read from `setup.py`) if the repo is a library released to PyPI

The two can be used at the same time.

Comment (fridex) [reply CONTRIBUTING.md:32]: BTW we could introduce a check to make sure they are in sync (requirements.txt corresponds to Pipfile):

Related: https://github.com/thoth-station/kebechet/issues/705

File: CONTRIBUTING.md (hunk 1 above)
Comment (goern) [reply CONTRIBUTING.md:51]: ping

Comment (fridex) [reply CONTRIBUTING.md:51]: Requiring external developers to have a proper dev setup when they want to contribute might be tricky or deterrent (given various versions of tools/linters available, their varying output across versions and different configurations available). Maybe just mentioning that each PR is checked on the CI side that has proper configuration might be better for new contributors, so they check results reported to the PR.

**Pull request merged**